datumaro/plugins/transforms.py

```diff
@@ -564,6 +567,66 @@ def transform_item(self, item):
                 annotations.append(ann.wrap())
         return item.wrap(annotations=annotations)
 
+class ProjectLabels(ItemTransform):
+    """
+    Changes the order of labels in the dataset from the existing
+    to the desired one and removes unknown labels. Updates or removes
+    the corresponding annotations.
+
+    Labels are matched by names (case dependent).
+
+    Useful for merging similar datasets.
+    """
+
+    def __init__(self, extractor: IExtractor, dst_labels: LabelCategories):
+        super().__init__(extractor)
+
+        self._categories = {}
+
+        src_label_cat = self._extractor.categories().get(AnnotationType.label)
+        self._make_label_id_map(src_label_cat, dst_labels)
+
+        src_mask_cat = self._extractor.categories().get(AnnotationType.mask)
+        if src_mask_cat is not None:
+            assert src_label_cat is not None
+            dst_mask_cat = MaskCategories(attributes=src_mask_cat.attributes)
+            dst_mask_cat.colormap = {
+                id: src_mask_cat[id]
```

Comment:
Shouldn't `id` be mapped here? Otherwise you're just recreating the source `MaskCategories` again.